### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by cap10morgan**
First version of a component for storing users' voting-method preferences and allowing default voting-method lookup by state (it pulls this from election-authority-works like voter-reg-works does for reg-methods).

[Pivotal card](https://www.pivotaltracker.com/story/show/106112456)

### CODE REVIEW

File: src/voting_method_works/states.clj [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+(defn voting-methods
+  "Returns the voting methods for `state` from the state election authority."
+  [state]
+  (let [authority (state-authority state)
+        methods (-> authority
+                    :ocd-division
+                    (nth 2)
+                    :voting-methods)]
+    (reduce #(merge %1 {(:type %2) (dissoc %2 :type)})
```
Comment (tank157) [inline src/voting_method_works/states.clj]: This is going to be problematic the way things are shaping up. It's possible for a state to have more than one voting-method of any given type. The prototypical case for this is that we still offer no-excuse absentee options for voters in WA and OR, but later on we might also have multiple in-person options too, from both early voting and in-person absentee.

Comment (cap10morgan) [inline src/voting_method_works/states.clj]: Yeah, I was thinking that once we have the "sub-type" key nailed down (which we decided on already, no?), I would add those as keys in nested maps under the `:type` keys.

Comment (tank157) [inline src/voting_method_works/states.clj]: There's not a specific single `sub-type` in what we currently have proposed. Let me paste an example here:

`#{{:type :by-mail, :primary false, :excuse-required false, :acceptable-forms #{{:name :wa_absentee}}} {:type :by-mail, :primary true, :excuse-required false} {:type :in-person, :instructions {:voting-id "Any person desiring to vote in person ...."}, :primary false, :excuse-required false}}`

So the combination of `:type`, `:primary`, and `:excuse-required` tell you a lot of what is necessary. In this case, the first in the set is the absentee by-mail case for WA, the second is the primary voting-method which is by-mail (indicative of all-mail), and the third is an in-person excuseless option (may need to check if this is still valid, I think it is, we had in-person id instructions so I assumed it was).

Comment (cap10morgan) [inline src/voting_method_works/states.clj]: Ah, right. Oh well, we can just put sets under the type keys. That makes lookups harder, but nothing we can't deal with.

Comment (tank157) [inline src/voting_method_works/states.clj]: When I run that example above, we lose the by-mail absentee option (though I don't know that on other orderings of data in the set we might not lose the all-mail option:

`{:by-mail {:primary true, :excuse-required false}, :in-person {:instructions {:voting-id "Any person desiring to vote in person at in Washington ..."}, :primary false, :excuse-required false}}`

**Comment by tank157** [comment]
LGTM

**Pull request merged**